File: ICARUS/Models/Icarus/Elements/CALLOUT.cs
using ICARUS.Models.Icarus.Elements.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ICARUS.Models.Icarus.Elements {

    /// <summary>
    /// A Bootstrap CALLOUT contains an icon, a header and a paragraph
    /// </summary>
    public class CALLOUT : Container {

        /// <summary>
        /// Jumbotron with image and header
        /// </summary>
        /// <param name="attributes"></param>
        public CALLOUT() : base("DIV", new MODEL()) {
            this.label = "CALLOUT";
        }

        /// <summary>
        /// Constructs an ARTICLE with the given id and label attributes already set
        /// </summary>
        /// <param name="formPost"></param>
        public CALLOUT(FormPost formPost) : base("DIV", formPost) {

        }
    }
}

File: ICARUS/Scripts/icarus/model/EL/container/callout/CALLOUT.js
/**
    Jumbotron with centered icon and text
*/
class CALLOUT extends CONTAINER {
    /**
        Constructs a Bootstrap Jumbotron.
        @param {CONTAINER} node The model
         @param {MODEL} model Object Model
     */
    constructor(node, model) {
        super(node, 'DIV', model);
        this.setClass('col-lg-4'); // Override icarus-container 

        this.body.pane.addClass('callout');


        this.icon = new GLYPHICON(this.body.pane, '', ICON.LOCK);
        if (model.dataId > 0) {            
            this.header = new HEADER(this.body.pane, new MODEL().set({
                'label': model.data.header
            }), 2);
            this.p = new P(this.body.pane, new MODEL(), model.data.p);
        } else {
            this.header = new HEADER(this.body.pane, new MODEL().set({
                'label': 'CALLOUT'
            }), 2);
            this.p = new P(this.body.pane, new MODEL(), 'Hello World');
        }
        

        this.populate(model.children);
    }
}